app/models/task.rb

```diff
@@ -0,0 +1,26 @@
+# == Schema Information
+#
+# Table name: tasks
+#
+#  id          :uuid             not null, primary key
+#  name        :string
+#  description :string
+#  state       :integer          default(0)
+#  project_id  :uuid
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+# Indexes
+#
+#  index_tasks_on_project_id  (project_id)
+#
+
+class Task < ActiveRecord::Base
+  belongs_to :project
+
+  enum state: [:todo, :in_progress, :done]
```

Comment:
It would probably be better to specify the states with specific integer values. Makes it easier to add enumerations with more values. See `Project` as an example:

```
enum state {
  todo: 10
  in_progress: 20, 
  done: 30
}
```

For instance if you insert a new enum value between todo and in_progress or between in_progress and done you'll need to ensure there is a migration for every value to move the values from 1->2 or 2->3 or 3->4.